dashboards-observability/public/components/explorer/home_table/history_table.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React, { useState, useRef, useEffect } from 'react';
+
+import {
+  EuiBasicTable,
+  EuiSpacer,
+  EuiLink,
+} from '@elastic/eui';
+
+
+interface TableData {
+  savedHistory: any;
+  savedQuerySearch: (searchQuery: string, selectedDateRange: [], selectedTimeStamp, selectedFields: []) => void;
+}
+
+export function Table(options: TableData) {
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+  const hisRef = useRef();
+  hisRef.current = pageIndex;
+  const thisRef = useRef();
+  thisRef.current = pageSize;
+
+
+  const onTableChange = ({ page = {} }) => {
+    const { index: pageIndex, size: pageSize } = page;
+
+    setPageIndex(pageIndex);
+    setPageSize(pageSize);
+  };
+
+  const columns = [
+    {
+      field: 'data',
+      name: 'Name',
```

Comment:
Question: Are we just showing the name in table? Or Are we adding Date Created and Date Modified to the table, like we have for notebooks and panels.

Reply:
No we're just displaying the name of the query. All other details are just passed internally for dispatch